overlord/ifacestate/handlers.go

```diff
@@ -755,6 +755,41 @@ func (m *InterfaceManager) doAutoConnect(task *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+// doDisconnectInterfaces creates tasks for disconnecting an interface and running its interface hooks.
+func (m *InterfaceManager) doDisconnectInterfaces(task *state.Task, _ *tomb.Tomb) error {
+	st := task.State()
+	st.Lock()
+	defer st.Unlock()
+
+	snapsup, err := snapstate.TaskSnapSetup(task)
+	if err != nil {
+		return err
+	}
+
+	snapName := snapsup.Name()
+	connections, err := m.repo.Connections(snapName)
+	if err != nil {
+		return err
+	}
+
+	connectts := state.NewTaskSet()
+	for _, connRef := range connections {
+		conn, err := m.repo.Connection(connRef)
+		if err != nil {
+			break
+		}
+		ts := DisconnectHooks(st, conn)
+		connectts.AddAll(ts)
+	}
+
+	task.SetStatus(state.DoneStatus)
+
+	snapstate.InjectTasks(task, connectts)
```

Comment:
seems we need some conflicts checks here, for example we cannot run the slot side hook on another snap if that snap is inactive

Reply:
Ok, I added check if inactive snap in the DisconnectHooks method, so that it's also effective on normal disconnect. I'm not yet clear about other conflicts, still pondering about this. I'll add test for this when it's more clear and agreed upon what to do.